inc/commonitilobject.class.php

```diff
@@ -6896,6 +6897,13 @@ function getTimelineItems() {
          ];
       }
 
+      //is requested from notification only public
+      if ($from_notification){
+         $restrict_fup = [
+            'is_private'   => 0,
```

Comment:
On this part (and below), you must check if the target have access to private items in its profile..
I suggest to replace the new function parameter by something like `get_private` and in notificationtargetcommonitilobject.class.php i remenber you have access to the private capability of the user *
See `$options['additionnaloption']['show_private']` usages (for followups/tasks) in the same `NotificationTargetCommonITILObject::getDataForObject`  (for followups)

Note this suggestion doesn't apply for document/validation where you use also your current parameter

Reply:
done